NewHaven/src/GameController.cpp

```diff
@@ -281,7 +281,7 @@ void GameController::shareTheWealth(){
         if(choice == 1){
             (*game_settings->players)[player_index]->buildVillage();
         }
-
+        setState(STANDBY);
         player_index = ++player_index % game_settings->players->size();
 
     }while(player_index != *current_turn_player && !game_settings->tracker->isEmpty());
```

Comment:
is empty check is only for moving onto the next player, so in a 2 game config this cannot be tested and during build village it will keep asking until the user decides to skip